manager/allocator/networkallocator/portallocator.go

```diff
@@ -285,7 +285,8 @@ func (pa *portAllocator) portsAllocatedInHostPublishMode(s *api.Service) bool {
 
 	if s.Spec.Endpoint != nil {
```

Comment:
> What if s.Spec.Endpoint is nil but s.Endpoint is not?

I do not think this is a possible scenario for host publish mode service, which this function is about.
Let me double check, and will get back on this.